services/data_sources/json_user_data_source.rb

```diff
@@ -172,24 +172,23 @@ def update_user!(user: nil)
       end
     end
 
-    def create_user!(id: nil, email: nil, password: nil, provider_credential: nil)
+    def create_user!(id: nil, email: nil, password: nil, provider_credentials: nil)
       users = self.users
       new_user = User.new(
         id: id,
         email: email,
         password_hash: BCrypt::Password.create(password),
-        provider_credentials: [provider_credential]
+        provider_credentials: provider_credentials
       )
 
       if !self.user_exist?(email: email)
         users.push(new_user)
         self.users = users
         logger.debug("Added user #{new_user.email}, writing out users.json to #{user_file_path}")
-        return new_user
       else
         logger.debug("Couldn't add user #{new_user.email} because they already exist")
-        return nil
       end
+      return new_user
```

Comment:
So in this case we always return a user, even if we didn't create one. Let's change it back to only returning the `new_user` if we actually created a user, otherwise return `nil` to signal that creation didn't happen.

Reply:
You're totally right